### PR TITLE
[Android Auto] Remove `MapboxNavigation` from `CarNavigationInfoObserver` constructor

### PR DESCRIPTION
**Pull request by kmadsen**
### Description

Addressing https://github.com/mapbox/mapbox-navigation-android/issues/6141

This change is a bit more involved so I'm still working on it. Opening as draft to share progress

 - [x] Remove `MapboxNavigation` from `CarNavigationInfoObserver` constructor
 - [x] Rename `CarNavigationInfoObserver` to `CarNavigationInfoProvider`
 - [x] Add unit tests for  `CarNavigationInfoProvider`
 - [x] Do manual testing

### CODE REVIEW

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationInfoProvider.kt [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    fun setNavigationInfo(): (NavigationTemplate.Builder).() -> Unit = {
+        with(carNavigationInfo.value) {
+            navigationInfo?.let {
+                setNavigationInfo(it)
+            }
+            destinationTravelEstimate?.let {
+                setDestinationTravelEstimate(it)
+            }
+        }
+    }
```
Comment (Zayankovsky) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationInfoProvider.kt:83]: Why not do something like this? Looks clearer in my opinion. 
```suggestion
    fun setNavigationInfo(templateBuilder: NavigationTemplate.Builder) {
        with(carNavigationInfo.value) {
            navigationInfo?.let {
                templateBuilder.setNavigationInfo(it)
            }
            destinationTravelEstimate?.let {
                templateBuilder.setDestinationTravelEstimate(it)
            }
        }
    }
```
Usage will look like this:
```kotlin
apply { navigationInfoProvider.setNavigationInfo(this) }
```
Also lambda will be optimized with my suggestion, but not with yours.

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationInfoProvider.kt:83]: was looking to optimize this interface for building your own `Screen`, so when you build the NavigationTemplate it looks like this

``` kotlin
return NavigationTemplate.Builder()
...
    .apply(navigationInfoProvider.setNavigationInfo())
    .build()
```

using `fun setNavigationInfo(templateBuilder: NavigationTemplate.Builder)` works too
``` kotlin
return NavigationTemplate.Builder()
..
    .also { navigationInfoProvider.setNavigationInfo(it) }
    .build()
```

One thing i don't like about `fun setNavigationInfo(): (NavigationTemplate.Builder).() -> Unit = {` is that it does not care at all about Java

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationInfoProvider.kt:83]: Also like this solution to be honest, as an extension function

``` kotlin
val builder = NavigationTemplate.Builder()
...
    .setCarNavigationInfo(navigationInfoProvider)
    .build()
```

``` kotlin
fun NavigationTemplate.Builder.setCarNavigationInfo(provider: CarNavigationInfoProvider) = apply {
    with(provider.carNavigationInfo.value) {
        navigationInfo?.let {
            setNavigationInfo(it)
        }
        destinationTravelEstimate?.let {
            setDestinationTravelEstimate(it)
        }
    }
}
```

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/CarNavigationInfoProvider.kt]: going with your suggestion

**Pull request merged**